### PR TITLE
Add HDFS operation jmx stats

### PR DESCRIPTION
**Pull request by weijiii**
## Description
* From #1971, add stats to operations called by `HdfsFileSystem`
* From #15260, add time stats for opening files

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java [hunk 1]
```diff
@@ -175,13 +175,13 @@ static boolean containsHiddenPathPartAfterIndex(String pathString, int startFrom
...
-        private final NamenodeStats namenodeStats;
+        private final HdfsNamenodeStats namenodeStats;
         private final RemoteIterator<TrinoFileStatus> fileStatusIterator;
 
-        private FileStatusIterator(Table table, Path path, FileSystem fileSystem, DirectoryLister directoryLister, NamenodeStats namenodeStats, boolean recursive)
+        private FileStatusIterator(Table table, Path path, FileSystem fileSystem, DirectoryLister directoryLister, HdfsNamenodeStats namenodeStats, boolean recursive)
         {
             this.path = path;
-            this.namenodeStats = namenodeStats;
+            this.namenodeStats= namenodeStats;
```
Comment (electrum) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java:184]: Formatting

**Review by electrum — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java (hunk 1 above)
Comment (weijiii) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java]: Done

File: lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystemFactory.java [hunk 2]
```diff
@@ -27,6 +28,7 @@
         implements TrinoFileSystemFactory
 {
     private final HdfsEnvironment environment;
+    private static final TrinoHdfsFileSystemStats STATS = new TrinoHdfsFileSystemStats();
```
[the remark this answers is not in the document]
Comment (weijiii) [reply lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystemFactory.java]: Thanks for taking a look :) Do you mean separate stats per `HdfsFileSystem` instance? In this case we don't have to make `TrinoHdfsFileSystemStats` singleton?

**Comment by weijiii** [comment]
Force-pushed to rebase.

**Comment by weijiii** [comment]
@raunaqmorarka Can you give a review on this PR? Also can I re-run this failing task? The error does not look related. Thanks.
```
Error: LinkageError occurred while loading main class io.trino.tests.product.launcher.cli.Launcher
	java.lang.UnsupportedClassVersionError: io/trino/tests/product/launcher/cli/Launcher has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0
```

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java [hunk 3]
```diff
@@ -125,6 +123,10 @@ public void configure(Binder binder)
         newExporter(binder).export(TrinoFileSystemCacheStats.class)
                 .as(generator -> generator.generatedNameOf(io.trino.plugin.hive.fs.TrinoFileSystemCache.class));
 
+        binder.bind(HdfsNamenodeStats.class).toInstance(HdfsFileSystemFactory.getNamenodeStats());
```
Comment (electrum) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java:126]: We can keep this bound as a singleton, like it was before

**Review by electrum — COMMENTED** [review]
Please rebase and then I'll merge this

**Comment by weijiii** [comment]
@electrum I've rebased and bound the stats to singleton. The test failure seems unrelated and it looks like we have an open issue on this #16315 . Please take a look. Thanks.
```
TestHiveTransactionalTable > testReadFullAcidPartitioned [groups: hive_transactional]
java.sql.SQLException: Query failed (#20230506_235422_01464_bx6fp): Found file in sub-directory of ACID directory: hdfs://hadoop-master:9000/user/hive/warehouse/test_read_full_acid_true_none_p24qoeaen9/part_col=2/delete_delta_0000002_0000005/delete_delta_0000002_0000005/bucket_00000

```

**Comment by weijiii** [comment]
Force pushed to squash commits

**Pull request merged**